packages/compiler/src/core/checker.ts

```diff
@@ -1753,17 +1770,31 @@ export function createChecker(program: Program): Checker {
     });
   }
 
-  function getSymbolLinks(s: Sym): SymbolLinks {
+  function getSymbolLinks(s: Sym, node?: Node): SymbolLinks {
```

Comment:
you should technically be able to get the node from the symbol `s.declarations[0]`

Reply:
or is it used for the diagnostic target? Think it might be unexpected that `getSymbolLinks` report deprecated errors

Reply:
It was a convenient place to add the check without having to complicate a lot of other calling code, though I admit it's a little cheesy.  Any alternative ideas?

Reply:
And yeah, it's for the diagnostic target, otherwise the alias usage doesn't get marked, the alias declaration does.

Reply:
so wondering related to the above comment on where the data should be. When resolving a `TypeReference` that's when we'd want ot report a diagnostic I think? so we could just maybe have the logic be standardized there and there we check the data on the target node.

Reply:
Yep, sounds good.  I think I can simplify this.